iroh-net/src/magicsock/endpoint/best_addr.rs

```diff
@@ -0,0 +1,202 @@
+        if let Some(addr) = self.addr() {
+            self.0 = None;
+            debug!(?reason, ?has_derp, old_addr = %addr, "remove best_addr");
+            // no longer relying on the direct connection
+            inc!(MagicsockMetrics, num_direct_conns_removed);
+            if has_derp {
+                // we are now relying on the relay connection, add a relay conn
+                inc!(MagicsockMetrics, num_relay_conns_added);
+            }
+            true
+        } else {
+            false
+        }
+    }
+
+    pub fn clear_if_equals(
+        &mut self,
+        addr: SocketAddr,
+        reason: ClearReason,
+        has_derp: bool,
+    ) -> bool {
+        match &self.addr() {
+            Some(best_addr) if *best_addr == addr => self.clear(reason, has_derp),
+            _ => false,
+        }
+    }
+
+    pub fn clear_trust(&mut self) {
+        if let Some(state) = self.0.as_mut() {
+            state.trust_until = None;
+        }
+    }
+
+    pub fn insert_if_better_or_reconfirm(
+        &mut self,
+        addr: SocketAddr,
+        latency: Option<Duration>,
+        source: Source,
+        confirmed_at: Instant,
+        has_derp: bool,
+    ) {
+        match self.0.as_mut() {
+            None => self.insert(addr, latency, source, confirmed_at, has_derp),
+            Some(state) => {
+                let candidate = AddrLatency { addr, latency };
+                if candidate.is_better_than(&state.addr) {
+                    self.insert(addr, latency, source, confirmed_at, has_derp)
+                } else if state.addr.addr == addr {
+                    state.confirmed_at = confirmed_at;
+                    state.trust_until = Some(source.trust_until(confirmed_at));
+                }
+            }
+        }
+    }
+
+    pub fn insert(
+        &mut self,
+        addr: SocketAddr,
+        latency: Option<Duration>,
+        source: Source,
+        confirmed_at: Instant,
+        has_derp: bool,
+    ) {
+        let trust_until = source.trust_until(confirmed_at);
+
+        info!(
+           %addr,
+           latency = ?latency,
+           trust_for = ?trust_until.duration_since(Instant::now()),
+           "new best_addr (candidate address with most recent pong)"
+        );
+        let was_empty = self.is_empty();
+        let inner = BestAddrInner {
+            addr: AddrLatency { addr, latency },
+            trust_until: Some(trust_until),
+            confirmed_at,
+        };
+        self.0 = Some(inner);
+        if was_empty && has_derp {
+            // we now have a direct connection, adjust direct connection count
+            inc!(MagicsockMetrics, num_direct_conns_added);
+            if has_derp {
+                // we no longer rely on the relay connection, decrease the relay connection
+                // count
+                inc!(MagicsockMetrics, num_relay_conns_removed);
+            }
+        }
+    }
+
+    pub fn state(&self, now: Instant) -> State {
```

Comment:
since the best address is per peer, these logs are probably more useful in the caller (where you already have similar logs)

Reply:
yup removed them here